### PR TITLE
test(draggable): convert tests from Cypress to Playwright - FE-6115

### PR DESCRIPTION
**Pull request by DipperTheDan**
### Proposed behaviour

Convert test to Playwright.

### Current behaviour

Tests are currently using Cypress.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

N/A

### Testing instructions

N/A

### CODE REVIEW

File: playwright/components/draggable/index.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import type { Page } from "@playwright/test";
+import { DRAGGABLE_ITEM, DRAG_ICON } from "./locators";
+
+export const draggableItem = (page: Page, text: string) =>
```
Comment (tomdavies73) [inline playwright/components/draggable/index.ts:4]: **question**: is there a better way we can do this, even just by adding some sort of tag, as this makes me genuinely scared

File: src/components/draggable/components.test-pw.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+} from "./draggable-container.component";
+import DraggableItem from "./draggable-item.component";
+import { Checkbox } from "../checkbox";
+import Textbox from "../textbox";
+import Box from "../box";
+
+export const DraggableDefault = () => {
+  const handleUpdate = (items?: (string | number | undefined)[]) => {
+    // eslint-disable-next-line no-console
+    console.log(items);
```
Comment (tomdavies73) [inline src/components/draggable/components.test-pw.tsx:13]: **question**: can we remove this, and the comment above?

**Review by tomdavies73 — DISMISSED** [review]

File: src/components/draggable/draggable.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  ([
+    ["One", 1],
+    ["One", 2],
+    ["One", 3],
+    ["One", 4],
+    ["Two", 3],
+    ["Three", 1],
+    ["Four", 2],
+  ] as const).forEach(([record, destinationId]) => {
+    test(`should drag ${record} and re-order DraggableItem to the ${destinationId} position`, async ({
```
Comment (stephenogorman) [inline src/components/draggable/draggable.pw.tsx:25]:
```suggestion
test(`should drag Draggable Label ${record} to position ${destinationId}`

File: src/components/draggable/draggable.pw.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      mount,
+      page,
+    }) => {
+      await mount(<DraggableCustom />);
+
+      const draggableItemElementRecord = await draggableItem(page, record);
```
Comment (stephenogorman) [inline src/components/draggable/draggable.pw.tsx:31]:
```suggestion
      const draggableItemElementRecord = draggableItem(page, record);

File: src/components/draggable/draggable.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      const draggableItemByPositionElementDestinationId = await draggableItemByPosition(
```
Comment (stephenogorman) [inline src/components/draggable/draggable.pw.tsx:32]:
```suggestion
      const draggableItemByPositionElementDestinationId = draggableItemByPosition(

File: src/components/draggable/draggable.pw.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        page,
+        destinationId
+      );
+      await draggableItemElementRecord.dragTo(
+        draggableItemByPositionElementDestinationId
```
Comment (stephenogorman) [inline src/components/draggable/draggable.pw.tsx:37]: There seems to be an issue here where the 1st and 2nd positions are mixed up. Each test which should drag the item to position 1 drags it to position 2 and vice versa. Or at least that's how it appears to me when I run in browser!

File: src/components/draggable/draggable.pw.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    });
+  });
+
+  test("should not change the order of the DraggableItems when dragged outside of the DraggableContainer", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DraggableDifferentContainers />);
+
+    const draggableItemElementFive = await draggableItem(page, "Five");
```
Comment (stephenogorman) [inline src/components/draggable/draggable.pw.tsx:52]: The suggestion regarding unnecessary 'awaits' when declaring constants that I included in the first test applies to this and all subsequent tests.

File: src/components/draggable/draggable.pw.tsx [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    const draggableItemElementSeven = await draggableItem(page, "Seven");
+    await draggableItemElementSeven.dragTo(draggableItemElementFive);
+
+    await expect(callbackCount).toBe(1);
+  });
+});
+
+test.describe("Accessibility tests for Draggable component", () => {
+  test("should pass accessibility tests for Draggable default", async ({
```
Comment (stephenogorman) [inline src/components/draggable/draggable.pw.tsx:106]:
```suggestion
  test("should pass accessibility tests for default example"

File: src/components/draggable/draggable.pw.tsx [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+    mount,
+    page,
+  }) => {
+    await mount(<DraggableDefault />);
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for Draggable with different containers", async ({
```
Comment (stephenogorman) [inline src/components/draggable/draggable.pw.tsx:115]:
```suggestion
test("should pass accessibility tests with different containers"

**Review by stephenogorman — CHANGES_REQUESTED** [review]

File: src/components/draggable/draggable.pw.tsx (hunk 6 above)
Comment (DipperTheDan) [reply src/components/draggable/draggable.pw.tsx:37]: Well what a bug that was. It existed already in Cypress and was caused by [this](https://github.com/Sage/carbon/blob/master/cypress/locators/draggable/index.js#L21). I've decided to re-write it to be a bit more robust.

**Review by stephenogorman — APPROVED** [review]

**Review by tomdavies73 — APPROVED** [review]

**Pull request merged**